haskell-src/Concordium/Crypto/ByteStringHelpers.hs

```diff
@@ -113,9 +114,9 @@ instance AE.ToJSON Short65K where
 -- |JSON instances based on base16 encoding.
 instance AE.FromJSON Short65K where
   parseJSON = AE.withText "Short65K" $ \t ->
-    let (bs, rest) = BS16.decode (Text.encodeUtf8 t)
-    in if BS.null rest then return (Short65K (BSS.toShort bs))
-       else AE.typeMismatch "Not a valid Base16 encoding." (AE.String t)
+    case BS16.decode (Text.encodeUtf8 t) of
+      Right bs -> return (Short65K (BSS.toShort bs))
+      Left _ -> AE.typeMismatch "Not a valid Base16 encoding." (AE.String t)
```

Comment:
For debugging it would be useful to keep the error.

Reply:
when needed it can be changed